.build/docker/run-tests.sh

```diff
@@ -0,0 +1,259 @@
+    # use it if we got a valid number (despite retry settings the curl above can still fail
+    [[ ${fetched_jenkins_executors} =~ '^[0-9]+$' ]] && jenkins_executors=${fetched_jenkins_executors}
+fi
+
+# find host's available cores and mem
+cores=1
+command -v nproc >/dev/null 2>&1 && cores=$(nproc --all)
+mem=1
+# linux
+command -v free >/dev/null 2>&1 && mem=$(free -b | grep Mem: | awk '{print $2}')
+# macos
+sysctl -n hw.memsize >/dev/null 2>&1 && mem=$(sysctl -n hw.memsize)
+
+# for relevant test targets calculate how many docker containers we should split the test list over
+case ${target} in
+    # test-burn doesn't have enough tests in it to split beyond 8, and burn and long we want a bit more resources anyway
+    "stress-test" | "fqltool-test" | "microbench" | "test-burn" | "long-test" | "cqlsh-test" )
+        [[ ${mem} -gt $((5 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "tests require minimum docker memory 6g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        docker_runs=1
+    ;;
+    "dtest" | "dtest-novnode" | "dtest-offheap" | "dtest-large" | "dtest-large-novnode" | "dtest-upgrade" )
+        [ -f "${cassandra_dtest_dir}/dtest.py" ] || { echo >&2 "${cassandra_dtest_dir}/dtest.py must exist"; exit 1; }
+        [[ ${mem} -gt $((15 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "dtests require minimum docker memory 16g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        docker_runs=1
+        test_script="dtest-python.sh"
+        docker_mounts="${docker_mounts} -v ${cassandra_dtest_dir}:/home/cassandra/cassandra-dtest"
+        # check that exists ${cassandra_dtest_dir}
+        [ -f "${cassandra_dtest_dir}/dtest.py" ] || { echo >&2 "${cassandra_dtest_dir}/dtest.py not found. please specify 'cassandra_dtest_dir' to point to the local cassandra-dtest source"; exit 1; }
+    ;;
+    "test"| "test-cdc" | "test-compression" | "jvm-dtest" | "jvm-dtest-upgrade")
+        [[ ${mem} -gt $((5 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "tests require minimum docker memory 6g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        max_docker_runs_by_cores=$( echo "sqrt( ${cores} / ${jenkins_executors} )" | bc )
+        max_docker_runs_by_mem=$(( ${mem} / ( 5 * 1024 * 1024 * 1024 * ${jenkins_executors} ) ))
+        docker_runs=$(( ${max_docker_runs_by_cores} < ${max_docker_runs_by_mem} ? ${max_docker_runs_by_cores} : ${max_docker_runs_by_mem} ))
+        docker_runs=$(( ${docker_runs} < 1 ? 1 : ${docker_runs} ))
+    ;;
+    *)
+    echo "unrecognized \"${target}\""
+    exit 1
+    ;;
+esac
+
+docker_runs=1 # tmp FIXME
+
+# Break up the requested split chunk into a number of concurrent docker runs, as calculated above
+# This will typically be between one to four splits. Five splits would require >25 cores and >25GB ram
+inner_splits=$(( $(echo $split_chunk | cut -d"/" -f2 ) * ${docker_runs} ))
```

Comment:
This was a bit hard for me to parse and follow. Could refactor it into something like the following as you see fit:

```
# Extract the numerator and denominator from the split_chunk variable
numerator=$(echo $split_chunk | cut -d"/" -f1)
denominator=$(echo $split_chunk | cut -d"/" -f2)

# Calculate inner_splits and inner_split_first using the extracted values
inner_splits=$((denominator * docker_runs))
inner_split_first=$(((numerator * docker_runs) - (docker_runs - 1)))

# Calculate docker_cpus with a scale of 2 decimal places
docker_cpus=$(echo "scale=2; ${cores} / (${jenkins_executors} * ${docker_runs})" | bc)
```